zip-0226.rst

```diff
@@ -1,8 +1,313 @@
+- The witness to the value base point, as defined in the `value base`_ equation is the auxiliary input :math:`\mathsf{AssetId}`.
+
+**Enforce Secure Identifier for Split Actions:** the following constraints must be added to prevent senders from changing the Asset identifier for the output note in a Split Action:
+
+- The Value Commitment Integrity should be changed
+    - Replace the input note value by a generic value, `v'`, as :math:`\mathsf{cv^net} = \mathsf{ValueCommit_rcv^OrchardZSA(v’ - v^new, AssetId)}`
+- Add a boolean ``split`` variable as an auxiliary witness. This variable is to be activated ``split = 1`` if the Action in question is a split and ``split = 0`` if the Action is actually spending an input note:
+    - If ``split = 1`` then set :math:`\mathsf{v}' = 0` otherwise :math:`\mathsf{v}'=\mathsf{v^{old}}` from the auxiliary input.
+- The Merkle Path Validity should check the existence of the note commitment as usual (and not like with dummy notes):
+    - Check that (path, pos) is a valid Merkle path of depth :math:`\mathsf{MerkleDepth^Orchard}`, from :math:`\mathsf{cm^old}` to the anchor :math:`\mathsf{rt^Orchard}`.
+- The Nullifier Integrity will be changed to prevent the identification of notes
+    - Replace the :math:`\psi_{old}` value with a generic :math:`\psi'` as :math:`\mathsf{nf_old = DeriveNullifier_nk}(\rho^\mathsf{old}, \psi', \mathsf{cm^old})`
+    - if :math:`split = 0` then constrain :math:`\psi' = \psi^{old}`. (Otherwise :math:`\psi'` should be sampled randomly.) 
+
+**Enabling Backwards Compatibility with ZEC Notes:** the following constraints must be added to enable backwards compatibility with the Orchard ZEC notes.
+
+The old note commitment is computed using a “rolling-aggregate” Sinsemilla commitment. This means that the commitment is computed by adding new chunks or windows to the accumulated value. This method will be used in order to maintain a single commitment instance for the old note commitment, that will be used both for Orchard ZEC notes and for ZSA notes. The original Orchard ZEC notes will be conserved and not actually be converted into ZSA notes, as we will always need to compute them.
+
+- The input note in the old note commitment integrity must either include an Asset identifier (ZSA note) or not (ZEC-Orchard note)
+    - If the Asset identifier auxiliary input is set :math:`\mathsf{AssetId}` = :math:`\mathcal{V}^\mathsf{Orchard}`
+        - NoteCommitment has a “compatibility” path that computes the note commitment as in plain Orchard (i.e.: without including the identifier)
+        - This path also uses the original domain separator for ZEC note commitment
+    - Else, 
+        - The NoteCommitment adds the identfier, :math:`\mathsf{AssetId}`, as a final “chunk” of the Sinsemilla commitment
+        - The NoteCommitment uses a different domain separator for ZSA note commitment
```

Comment:
This should be moved either to the top of the "Circuit Statement" section, or more likely into a separate section before it describing changes to the definition of Sinsemilla (or the Sinsemilla gadget in Halo 2 terms). Then it can be referenced in the asset ID changes above.

Reply:
This has been rewritten